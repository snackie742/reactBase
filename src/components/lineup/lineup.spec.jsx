import React from 'react';
import { shallow } from 'enzyme';
import Lineup from './lineup';

const defaultProps = {
    playerEntry: {
        player: { FirstName: 'Player', LastName: 'LastName'},
        stats: {
            AtBats: {},
            Runs: {},
            Hits: {},
            BatterWalks: {},
            BatterStrikeouts: {},
            BattingAvg: {},
        },
    },
};

const setup = (propOverrides) => {
  const props = Object.assign({}, defaultProps, propOverrides);
  const wrapper = shallow(<Lineup {...props} />);
  const component = wrapper.instance();
  return { wrapper, component };
};

describe('Lineup', () => {
  describe('<Lineup/>', () => {
    describe('(Snapshot)', () => {
      test('renders as expected', () => {
          defaultProps.playerEntry.stats.AtBats['#text'] = '4';
          defaultProps.playerEntry.stats.Runs['#text'] = '2';
          defaultProps.playerEntry.stats.Hits['#text'] = '3';
          defaultProps.playerEntry.stats.BatterWalks['#text'] = '0';
          defaultProps.playerEntry.stats.BatterStrikeouts['#text'] = '1';
          defaultProps.playerEntry.stats.BattingAvg['#text'] = '0.750';
        const { wrapper } = setup(defaultProps);
        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
