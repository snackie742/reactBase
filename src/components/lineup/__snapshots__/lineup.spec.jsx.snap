// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lineup <Lineup/> (Snapshot) renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    playerEntry={
        Object {
            "player": Object {
              "FirstName": "Player",
              "LastName": "LastName",
            },
            "stats": Object {
              "AtBats": Object {
                "#text": "4",
              },
              "BatterStrikeouts": Object {
                "#text": "1",
              },
              "BatterWalks": Object {
                "#text": "0",
              },
              "BattingAvg": Object {
                "#text": "0.750",
              },
              "Hits": Object {
                "#text": "3",
              },
              "Runs": Object {
                "#text": "2",
              },
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
>
        <div
                className="col-lg-5"
        >
                Player LastName
        </div>
        <div
                className="col-lg-6"
        >
                <div
                        className="row"
                >
                        <div
                                className="col lineupEntry"
                        >
                                 
                                4
                        </div>
                        <div
                                className="col lineupEntry"
                        >
                                 
                                2
                        </div>
                        <div
                                className="col lineupEntry"
                        >
                                 
                                3
                        </div>
                        <div
                                className="col lineupEntry"
                        >
                                 
                                0
                        </div>
                        <div
                                className="col lineupEntry"
                        >
                                 
                                1
                        </div>
                        <div
                                className="col lineupEntry"
                        >
                                 
                                0.750
                        </div>
                </div>
        </div>
</div>,
      "className": "col-lg lineupCard",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-lg-5"
>
            Player LastName
</div>,
          <div
            className="col-lg-6"
>
            <div
                        className="row"
            >
                        <div
                                    className="col lineupEntry"
                        >
                                     
                                    4
                        </div>
                        <div
                                    className="col lineupEntry"
                        >
                                     
                                    2
                        </div>
                        <div
                                    className="col lineupEntry"
                        >
                                     
                                    3
                        </div>
                        <div
                                    className="col lineupEntry"
                        >
                                     
                                    0
                        </div>
                        <div
                                    className="col lineupEntry"
                        >
                                     
                                    1
                        </div>
                        <div
                                    className="col lineupEntry"
                        >
                                     
                                    0.750
                        </div>
            </div>
</div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Player LastName",
            "className": "col-lg-5",
          },
          "ref": null,
          "rendered": "Player LastName",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
>
              <div
                            className="col lineupEntry"
              >
                             
                            4
              </div>
              <div
                            className="col lineupEntry"
              >
                             
                            2
              </div>
              <div
                            className="col lineupEntry"
              >
                             
                            3
              </div>
              <div
                            className="col lineupEntry"
              >
                             
                            0
              </div>
              <div
                            className="col lineupEntry"
              >
                             
                            1
              </div>
              <div
                            className="col lineupEntry"
              >
                             
                            0.750
              </div>
</div>,
            "className": "col-lg-6",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="col lineupEntry"
>
                   
                  4
</div>,
                <div
                  className="col lineupEntry"
>
                   
                  2
</div>,
                <div
                  className="col lineupEntry"
>
                   
                  3
</div>,
                <div
                  className="col lineupEntry"
>
                   
                  0
</div>,
                <div
                  className="col lineupEntry"
>
                   
                  1
</div>,
                <div
                  className="col lineupEntry"
>
                   
                  0.750
</div>,
              ],
              "className": "row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "4",
                  ],
                  "className": "col lineupEntry",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "4",
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "2",
                  ],
                  "className": "col lineupEntry",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "2",
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "3",
                  ],
                  "className": "col lineupEntry",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "3",
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "0",
                  ],
                  "className": "col lineupEntry",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "0",
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "1",
                  ],
                  "className": "col lineupEntry",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "1",
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "0.750",
                  ],
                  "className": "col lineupEntry",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "0.750",
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
>
          <div
                    className="col-lg-5"
          >
                    Player LastName
          </div>
          <div
                    className="col-lg-6"
          >
                    <div
                              className="row"
                    >
                              <div
                                        className="col lineupEntry"
                              >
                                         
                                        4
                              </div>
                              <div
                                        className="col lineupEntry"
                              >
                                         
                                        2
                              </div>
                              <div
                                        className="col lineupEntry"
                              >
                                         
                                        3
                              </div>
                              <div
                                        className="col lineupEntry"
                              >
                                         
                                        0
                              </div>
                              <div
                                        className="col lineupEntry"
                              >
                                         
                                        1
                              </div>
                              <div
                                        className="col lineupEntry"
                              >
                                         
                                        0.750
                              </div>
                    </div>
          </div>
</div>,
        "className": "col-lg lineupCard",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-lg-5"
>
              Player LastName
</div>,
            <div
              className="col-lg-6"
>
              <div
                            className="row"
              >
                            <div
                                          className="col lineupEntry"
                            >
                                           
                                          4
                            </div>
                            <div
                                          className="col lineupEntry"
                            >
                                           
                                          2
                            </div>
                            <div
                                          className="col lineupEntry"
                            >
                                           
                                          3
                            </div>
                            <div
                                          className="col lineupEntry"
                            >
                                           
                                          0
                            </div>
                            <div
                                          className="col lineupEntry"
                            >
                                           
                                          1
                            </div>
                            <div
                                          className="col lineupEntry"
                            >
                                           
                                          0.750
                            </div>
              </div>
</div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Player LastName",
              "className": "col-lg-5",
            },
            "ref": null,
            "rendered": "Player LastName",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="row"
>
                <div
                                className="col lineupEntry"
                >
                                 
                                4
                </div>
                <div
                                className="col lineupEntry"
                >
                                 
                                2
                </div>
                <div
                                className="col lineupEntry"
                >
                                 
                                3
                </div>
                <div
                                className="col lineupEntry"
                >
                                 
                                0
                </div>
                <div
                                className="col lineupEntry"
                >
                                 
                                1
                </div>
                <div
                                className="col lineupEntry"
                >
                                 
                                0.750
                </div>
</div>,
              "className": "col-lg-6",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="col lineupEntry"
>
                     
                    4
</div>,
                  <div
                    className="col lineupEntry"
>
                     
                    2
</div>,
                  <div
                    className="col lineupEntry"
>
                     
                    3
</div>,
                  <div
                    className="col lineupEntry"
>
                     
                    0
</div>,
                  <div
                    className="col lineupEntry"
>
                     
                    1
</div>,
                  <div
                    className="col lineupEntry"
>
                     
                    0.750
</div>,
                ],
                "className": "row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "4",
                    ],
                    "className": "col lineupEntry",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "4",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "2",
                    ],
                    "className": "col lineupEntry",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "2",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "3",
                    ],
                    "className": "col lineupEntry",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "3",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "0",
                    ],
                    "className": "col lineupEntry",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "0",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "1",
                    ],
                    "className": "col lineupEntry",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "1",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "0.750",
                    ],
                    "className": "col lineupEntry",
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "0.750",
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
